pkg/virt-controller/watch/drain/evacuation/evacuation.go

```diff
@@ -388,7 +395,20 @@ func (c *EvacuationController) sync(node *k8sv1.Node, vmisOnNode []*virtv1.Virtu
 		}
 		return nil
 	}
-	freeSpots := maxParallelMigrations - len(activeMigrations)
+
+	maxParallelMigrationsPerOutboundNode :=
+		int(*c.clusterConfig.GetMigrationConfiguration().ParallelOutboundMigrationsPerNode)
+	if activeMigrationsFromThisSourceNode >= maxParallelMigrationsPerOutboundNode {
+		if len(migrationCandidates) > 0 || len(nonMigrateable) > 0 {
+			c.Queue.AddAfter(node.Name, 5*time.Second)
+		}
+		return nil
+	}
+
+	// check how many concurrent running migrations from this source node
+	// calculate (maxPerOutbound - current running) take min and create VMIM for them
+	freeSpots := maxParallelMigrationsPerOutboundNode - activeMigrationsFromThisSourceNode
```

Comment:
So, just to make sure I got it right, correct me if I'm wrong:

Let's say that we have 2 nodes, each with 10 VMs, and `maxParallelMigrationsPerOutboundNode == 5`, and `maxParallelMigrationsPerCluster == 10`.

Before your changes if we had to evict these 2 nodes, what could happen is that the first node could create 10 VMIM objects. Then, 5 of these VMIMs would be running, and another 5 would be pending (because of [this](https://github.com/kubevirt/kubevirt/blob/v0.58.0/pkg/virt-controller/watch/migration.go#L870)). When the second node would get to this logic it wouldn't create any objects since 10 already exist.

It that accurate?

Reply:
@iholder-redhat Yes exactly what you've described.